### PR TITLE
Fix LICENSE file

### PR DESCRIPTION
**Pull request by jtattermusch**
See https://github.com/grpc/grpc-go/issues/1309

New LICENSE file is copied from http://www.apache.org/licenses/LICENSE-2.0.txt

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -171,13 +176,24 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (mehrdada) [inline LICENSE]: Similarly to the grpc/grpc version, shouldn't this template parameter be specialized?

Comment (carl-mastrangelo) [inline LICENSE]: Other OSS projects don't seem to:

https://github.com/kubernetes/kubernetes/blob/master/LICENSE
https://github.com/netty/netty/blob/4.1/LICENSE.txt

Comment (mehrdada) [inline LICENSE]: The Go repo has it though, so I am not sure who is doing it right. Whatever we do we need to be consistent among ourselves at least: https://github.com/grpc/grpc-go/blob/master/LICENSE

**Comment by jtattermusch** [comment]
Let's merge the current version as it is a significant improvement over what we have right now (and is consistent with grpc/grpc and grpc/grpc-go). Later we can remove the appendix.

File: LICENSE (hunk 1 above)
Comment (jtattermusch) [inline LICENSE]: grpc/grpc won't have it and I'm pretty sure go has been updated ad-hoc without doing much research. Let's merge this and update go later.

**Pull request merged**